capturable/src/iosMain/kotlin/dev/shreyaspatil/capturable/Capturable.ios.kt

```diff
@@ -0,0 +1,114 @@
+package dev.shreyaspatil.capturable
+
+import androidx.compose.ui.draw.CacheDrawModifierNode
+import androidx.compose.ui.graphics.asComposeCanvas
+import androidx.compose.ui.graphics.asComposeImageBitmap
+import androidx.compose.ui.graphics.drawscope.draw
+import androidx.compose.ui.graphics.drawscope.drawIntoCanvas
+import androidx.compose.ui.graphics.nativeCanvas
+import androidx.compose.ui.node.DelegatableNode
+import androidx.compose.ui.node.DelegatingNode
+import dev.shreyaspatil.capturable.controller.CaptureController
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.jetbrains.skia.Bitmap
+import org.jetbrains.skia.ColorAlphaType
+import org.jetbrains.skia.ColorInfo
+import org.jetbrains.skia.ColorSpace
+import org.jetbrains.skia.ColorType
+import org.jetbrains.skia.ImageInfo
+import org.jetbrains.skia.Picture
+import org.jetbrains.skia.PictureRecorder
+
+@Suppress("unused")
+actual class CapturableModifierNode actual constructor(controller: CaptureController) : DelegatingNode(), DelegatableNode {
+
+    private val currentController = MutableStateFlow(controller)
+
+    override fun onAttach() {
+        super.onAttach()
+        coroutineScope.launch {
+            observeCaptureRequestsAndServe()
+        }
+    }
+
+    actual fun updateController(controller: CaptureController) {
+        currentController.value = controller
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private suspend fun observeCaptureRequestsAndServe() {
+        currentController
+            .flatMapLatest { it.captureRequests }
+            .collect { request ->
+                val completable = request.imageBitmapDeferred
+                try {
+                    val picture = getCurrentContentAsPicture()
+                    val bitmap = withContext(Dispatchers.Default) {
+                        picture.asBitmap(request.config)
+                    }
+
+                    completable.complete(bitmap.asComposeImageBitmap())
+                } catch (error: Throwable) {
+                    completable.completeExceptionally(error)
+                }
+            }
+    }
+
+    private suspend fun getCurrentContentAsPicture(): Picture {
+        return drawCanvasIntoPicture()
+    }
+
+    private suspend fun drawCanvasIntoPicture(): Picture {
+        val pictureDrawn = CompletableDeferred<Unit>()
+        val pictureRecorder = PictureRecorder()
+        var picture = Picture.makePlaceholder(org.jetbrains.skia.Rect(0f, 0f, 0f, 0f))
+
+        val delegatedNode = delegate(
+            CacheDrawModifierNode {
+                val width = this.size.width
+                val height = this.size.height
+
+                onDrawWithContent {
```

Comment:
I would strongly recommend to use this API which is KMP-compatible and comes from `androix.compose` package which will be reliable: https://developer.android.com/develop/ui/compose/graphics/draw/modifiers#composable-to-bitmap

Reply:
I tried this. It says cmp but shows up in android only at the moment. Maybe some temporary doc/release bug. 
+ it's in alpha. So we couldn move to it when it's stable
What do say?

Reply:
Okay, works 👍🏻